schema/snapcraft.yaml

```diff
@@ -73,13 +73,31 @@ properties:
     type: string
     description: name of the part that provides source files that will be parsed to extract snap metadata information
   name:
-    type: string
     description: name of the snap package
-    validation-failure:
-      "{.instance!r} is not a valid snap name. Snap names consist of lower-case
-      alphanumeric characters and hyphens. They cannot be all numbers. They
-      also cannot start or end with a hyphen."
-    pattern: "^(?:[a-z0-9]+-?)*[a-z](?:-?[a-z0-9])*$"
+    allOf:
+      - type: string
+        # this failure message avoids printing repr of the thing, as it could be huge
+        validation-failure: "snap names need to be strings."
```

Comment:
snap names need to be strings _limited to a length of 40 character_.

Maybe consider something wrt the length?

Reply:
the length error doesn't use the `validation-failure` message, somehow.